### ISSUE TITLE
Rule S3881: message needs to be improved

### ISSUE THREAD

**Issue by Corniel** [SonarSource/sonar-dotnet#2339]
### Description
See the code below. I assume it is not related to issue #1728. As far as I know I followed all recommended guidelines, and just cleaned up some code (like comments), renamed a variable, and made Dispose() using an expression body (but that is not the reason it failed as far as I know).

### Repro steps
``` CSharp
using System;

namespace SonarAnalyzer.Experiments.CSharp
{
    public class S3881 : IDisposable //compliant
    {
        public S3881(IDisposable prop) => Property = prop;

        public IDisposable Property { get; }

        public void Dispose() => Dispose(true);

        protected virtual void Dispose(bool disposing)
        {
            if (!isDisposed)
            {
                if (disposing)
                {
                    Property.Dispose();
                }
                isDisposed = true;
            }
        }
        private bool isDisposed;
    }
}
```

### Expected Result
I would expect not to see a warning at all, obviously, but because this is a quite complex check, at least a hint on which part of the pattern is missing.

### Related information
* SonarC# Version 7.11.0.8083
* https://github.com/Corniel/SonarAnalyzer.Experiments/blob/master/src/SonarAnalyzer.Experiments.CSharp/S3881.cs

**Comment by andrei-epure-sonarsource** [comment]
According to [the rule description](https://rules.sonarsource.com/csharp/RSPEC-3881):

> The content of the Dispose() method should be invocation of Dispose(true) followed by GC.SuppressFinalize(this)

**Comment by Corniel** [comment]
@andrei-epure-sonarsource  It would be nice, if the error message is clear on what is missing. So in this case, the message is too vague. It should make clear that the `GC.SuppressFinalize(this)` is missing, or (in other cases) that `protected Dispose(true)` is lacking a virtual method, or something else.

Now, based on the message, it is not clear what to do.
